platform/files.go

```diff
@@ -83,8 +99,9 @@ type LayerMetadata struct {
 }
 
 type RunImageMetadata struct {
-	TopLayer  string `json:"topLayer" toml:"top-layer"`
-	Reference string `json:"reference" toml:"reference"`
+	TopLayer  string                   `json:"topLayer" toml:"top-layer"`
+	Reference string                   `json:"reference" toml:"reference"`
+	Target    buildpack.TargetMetadata `json:"target" toml:"target"`
```

Comment:
Thanks --  this is mostly addressed in [this commit](https://github.com/buildpacks/lifecycle/pull/994/commits/7314fa9df829ddc8dcd2f19bc802dcd5a6f62dd9) but i want to leave this as unresolved for a while longer since I'm expecting some further refinements to what I did :)